7_react/react1/week3/TodoList/src/components/TodoForm.tsx

```diff
@@ -0,0 +1,68 @@
+import "./todoForm.css";
+import { ChangeEvent, useState } from "react";
+import { Todo } from "../App";
+
+function TodoForm({
+  todos,
+  setTodos,
+}: {
+  todos: Todo[];
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+}) {
+  const [newTodoItem, setNewTodoItem] = useState<Todo>({
+    id: 0,
+    description: "",
+    deadline: "",
+  });
+
+  function generateUniqueId(): number {
+    return parseInt(
+      Date.now().toString(36) + Math.random().toString(36).substr(2),
+      36
+    );
+  }
+
+  const changeTodoInput = (e: ChangeEvent<HTMLInputElement>) => {
+    setNewTodoItem({ ...newTodoItem, description: e.target.value });
+  };
+  const changeDeadLineInput = (e: ChangeEvent<HTMLInputElement>) => {
+    setNewTodoItem({ ...newTodoItem, deadline: e.target.value });
+  };
+
+  const addNewTodo = () => {
```

Comment:
You could add a condition to check if the new item is empty before adding it to the list. This prevents empty items from being included.